src/swarm-tui/Swarm/TUI/Controller/Util.hs

```diff
@@ -117,3 +157,41 @@ resetViewport :: ViewportScroll Name -> EventM Name AppState ()
 resetViewport n = do
   vScrollToBeginning n
   hScrollToBeginning n
+
+-- | Modifies the game state using a fused-effect state action.
+zoomGameState :: (MonadState AppState m, MonadIO m) => Fused.StateC GameState (TimeIOC (Fused.LiftC IO)) a -> m a
+zoomGameState f = do
+  gs <- use gameState
+  (gs', a) <- liftIO (Fused.runM (runTimeIO (Fused.runState gs f)))
+  gameState .= gs'
+  return a
+
+onlyCreative :: (MonadState AppState m) => m () -> m ()
+onlyCreative a = do
+  c <- use $ gameState . creativeMode
+  when c a
+
+-- | Create a list of handlers with embedding events and using pattern matching.
+allHandlers ::
+  (Ord e2, Enum e1, Bounded e1) =>
+  (e1 -> e2) ->
+  (e1 -> (Text, EventM Name AppState ())) ->
+  [KeyEventHandler e2 (EventM Name AppState)]
+allHandlers eEmbed f = map handleEvent1 [minBound .. maxBound]
+ where
+  handleEvent1 e1 = let (n, a) = f e1 in onEvent (eEmbed e1) n a
+
+runBaseTerm :: (MonadState AppState m) => Maybe TSyntax -> m ()
+runBaseTerm = maybe (pure ()) startBaseProgram
+ where
+  -- The player typed something at the REPL and hit Enter; this
+  -- function takes the resulting ProcessedTerm (if the REPL
```

Comment:
```suggestion
  -- function takes the resulting term (if the REPL
```
Oops, looks like I forgot to remove the reference to `ProcessedTerm` here in #1928 .  Would you mind including this change?

Reply:
Could have also survived while I was rebasing, no problem. 🙂